### DISCUSSION TITLE
Redis script with SMEMBERS call should return empty if key doesn't exist

### DISCUSSION THREAD

**Discussion by FireMasterK** [dragonflydb/dragonfly#4371] [Q&A] [answered]
**Describe the bug**
When using `redis.call('SMEMBERS', '...')` on a key that doesn't exist, I get an error, which is different behavior compared to redis.

**To Reproduce**
1. Run a script using `redis.call('SMEMBERS', '...')` on a key that doesn't exist.

**Expected behavior**
The script shouldn't throw an error.

**Screenshots**

**Environment (please complete the following information):**
 - OS: [ubuntu 24.04]
 - Kernel: `6.8.0-50-generic #51-Ubuntu SMP PREEMPT_DYNAMIC Sat Nov  9 17:58:29 UTC 2024 x86_64 x86_64 x86_64` GNU/Linux
 - Containerized?: Docker
 - Dragonfly Version: Latest build - v1.26.0

**Reproducible Code Snippet**
I'm using redis-rs to run the script.
```
local abc = redis.call('SMEMBERS', 'test')
```

**Comment by romange** [comment]
what's the error you are getting?

**Comment by FireMasterK** [comment]
I got an error like this: `An error was signalled by the server - ResponseError: Error running script (call to 99f2a53b7a97df8baf906e38cc4f0e26e06a2676): @user_script:8: script tried accessing undeclared key, key: redacted`

**Comment by romange** [comment]
You should pre-declare the keys, see https://www.dragonflydb.io/docs/command-reference/generic/eval#syntax for more details.

**Comment by romange** [comment, marked as answer]
Please let me know if you have any more questions. There is more material around lua transactions in dragonfly.
See also https://www.dragonflydb.io/docs/managing-dragonfly/scripting . Passing `--default_lua_flags=allow-undeclared-keys` solves the issue but is unadvised because it hinders perfomance of Dragonfly around lua calls.

**Reply by FireMasterK** [reply]
Thanks, this is nice, this brings back the compatibility I was looking for with my application!
I have scripts in my application that can fetch additional keys based on arguments and the result of one, so this fixed it!
